archimedes/registry.py

```diff
@@ -0,0 +1,192 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2014-2018 Bitergia
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA 02111-1307, USA.
+#
+# Authors:
+#   Valerio Cosentino <valcos@bitergia.com>
+#
+
+import json
+import logging
+import os
+
+from archimedes.errors import NotFoundError, RegistryError
+from archimedes.meta_obj import MetaObj
+from archimedes.utils import load_json
+
+logger = logging.getLogger(__name__)
+
+REGISTRY_NAME = ".registry"
+
+
+class Registry:
+
+    def __init__(self, root_path):
+        """This class allows to manipulate the Kibana objects using aliases, thus avoiding the user
+        to deal with their IDs and titles. The registry is saved in the root folder of Archimedes,
+        with name .registry. The content of the registry is as following:
+        [
+            "1": {
+                    'id': 'Search:_pull_request:false',
+                    'title': 'Search:_pull_request:false',
+                    'type': 'search',
+                    'version': 1
+            },
+            "2": {
+                    'id': '8539ada0-9960-11e8-8771-a349686d998a',
+                    'title': 'dockerhub',
+                    'type': 'index-pattern',
+                    'version': 1
+            },
+            ...
+        ]
+
+        :param root_path: path where the registry will be stored
+        """
+        self.path = os.path.join(root_path, REGISTRY_NAME)
+
+        if not os.path.exists(self.path):
+            self.__create_registry()
+
+    def find(self, alias=None, obj_type=None):
+        """Return the information in the registry. If `alias` and `obj_type` are null, it returns the
+        content of all the registry. Otherwise, it returns the information related to the single `alias` or
+        the aliases with the given `obj_type`.
```

Comment:
Shouldn't be better to have two different methods? One to get the information of the whole registry and another one to get just one alias entry?